### ISSUE TITLE
Bogus "selector is not a named variable" on assumed-shape CLASS(*) array

### ISSUE THREAD

**Issue by klausler** [flang-compiler/f18#575]
```
subroutine s(x)
  class(*) :: x(:)
  select type (x)
  end select
end subroutine
```

**Comment by klausler** [comment]
Allocatable non-dummy arrays are broken too.

**Comment by jeanPerier** [comment]
Is fixed in Peter's PR #611.
Issue was that empty subscript triplets were inserted into whole array references in `ExpressionAnalyzer::Designate(DataRef &&ref)` (called while analyzing the selector) which later caused `UnwrapWholeSymbolDataRef` to fail.

**Comment by jeanPerier** [comment]
Fixed by PR #611.
